### PR TITLE
Implement generic function tests

### PR DESCRIPTION
**Pull request by kevindiu**
### Description:

This PR implemented the generic function tests removed in this commit.
https://github.com/vdaas/vald/pull/2005/commits/baf038d8b8f8dce9f24122ded40369f19e7e8624

### Versions:

- Go Version: 1.20.3
- Docker Version: 20.10.8
- Kubernetes Version: 1.22.0
- NGT Version: 2.0.9

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/main/CONTRIBUTING.md) document and completed [our CLA agreement](https://cla-assistant.io/vdaas/vald).
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?

### CODE REVIEW

File: internal/test/mock/server_stream.go [hunk 1]
```diff
@@ -17,7 +17,7 @@ import (
 	"context"
 
 	"github.com/vdaas/vald/apis/grpc/v1/payload"
-	"github.com/vdaas/vald/internal/net/grpc"
```
Comment (kevindiu) [inline internal/test/mock/server_stream.go:20]: to fix cycle import

File: internal/slices/slices_test.go [hunk 2]
```diff
@@ -13,4 +13,179 @@
 // limitations under the License.
 package slices
 
+import (
+	"errors"
```
[the remark this answers is not in the document]
Comment (kevindiu) [reply internal/slices/slices_test.go:17]:
```suggestion
	"internal/errors"
```

File: internal/net/grpc/stream_test.go [hunk 3]
```diff
@@ -18,12 +18,189 @@
...
+		return nil
+	}
+	tests := []test{
+		func() test {
+			insertCnt := 1
+			reqs, err := request.GenMultiInsertReq(request.Float, vector.Gaussian, insertCnt, f32VecDim, nil)
+			if err != nil {
+				t.Fatal(err)
+			}
+
```
Comment (vankichi) [inline internal/net/grpc/stream_test.go:99]:
```suggestion
```

File: internal/net/grpc/stream_test.go [hunk 4]
```diff
@@ -18,12 +18,189 @@
...
+				},
+			}
+		}(),
+		func() test {
+			insertCnt := 10
+			reqs, err := request.GenMultiInsertReq(request.Float, vector.Gaussian, insertCnt, f32VecDim, nil)
+			if err != nil {
+				t.Fatal(err)
+			}
+
```
Comment (vankichi) [inline internal/net/grpc/stream_test.go:118]:
```suggestion
```

File: internal/net/grpc/stream_test.go [hunk 5]
```diff
@@ -18,12 +18,189 @@
...
+		checkFunc  func([]*payload.Object_StreamLocation, want, error) error
+		beforeFunc func(*testing.T, args)
+		afterFunc  func(*testing.T, args)
+	}
+
+	const (
+		name               = "vald-agent-ngt-1" // agent name
+		f32VecDim          = 3                  // float32 vector dimension
+		ip                 = "localhost"        // ip address
+		defaultConcurrency = 10
```
Comment (vankichi) [inline internal/net/grpc/stream_test.go:57]:
```suggestion
		name        = "vald-agent-ngt-1" // agent name
		f32VecDim   = 3                  // float32 vector dimension
		ip          = "localhost"        // ip address
		concurrency = 10
```

**Review by vankichi — CHANGES_REQUESTED** [review]

File: internal/net/grpc/stream_test.go (hunk 5 above)
Comment (kevindiu) [reply internal/net/grpc/stream_test.go]: fixed

**Review by hlts2 — APPROVED** [review]
LGTM

**Review by vankichi — APPROVED** [review]
LGTM

**Pull request merged**